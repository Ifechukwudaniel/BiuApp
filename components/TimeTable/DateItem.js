import React from 'react';
import { Text} from 'native-base'
import {TouchableOpacity} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import { LinearGradient } from 'expo-linear-gradient';

const days= ['Mon', 'Tue', 'Wen', 'Thu', 'Fri', 'Sat', 'Sun'] 

const DateItem = ({active, date, day}) => {
    return (
    <TouchableOpacity style={styles.day}>
        <LinearGradient style={styles.day} colors={active? ['#EA3939', '#971313'] :['#126FFE', '#0B316C']}>
           <Text style={styles.dayDate}> {date} </Text>
           <Text  style={styles.dayMonth}> {days[day]} </Text>
        </LinearGradient>
    </TouchableOpacity>
    );
}
 
const styles = EStyleSheet.create({
    day:{
        width:"80rem",
        height:"80rem",
        borderRadius: '10rem',
        marginRight: '20rem',
        alignContent: 'center',
    },
   
    dayDate:{
        fontSize: '25rem', 
        color:"#fff",
        fontFamily:"Itim", 
        marginLeft:'20rem',
        paddingTop: '10rem',
    },
    dayMonth:{
        fontSize: '20rem', 
        color:"#fff",
        fontFamily:"Itim",  
        marginLeft:'18rem'
    }
})
export default DateItem;