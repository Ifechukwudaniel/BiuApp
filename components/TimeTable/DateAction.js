import React from 'react';
import {View, StatusBar, TouchableOpacity, ScrollView} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import { LinearGradient } from 'expo-linear-gradient';
import DateItem from './DateItem'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
import Ripple from 'react-native-material-ripple';
import { Audio } from 'expo-av';
import Header from '../Header'
import { Container,  Button, Text,  } from 'native-base'


const playSound=  async ()=>{
    const soundObject = new Audio.Sound();
    try {
        await soundObject.loadAsync(require('../../assets/sounds/beep.mp3'));
        await soundObject.playAsync();
        } catch (error) {
    }
}
 
const DateAction = ({}) => {
    return (
            <View style={styles.action}>
                <LinearGradient style={styles.gradient} colors={['#0F9A47', '#027831']}>
                       <View>
                           <Text style={styles.timeText}> 2:00pm</Text>
                           <Text style={styles.timeText}> 4:00pm</Text>
                       </View>
                        <View style={styles.line}/>
                      <View>
                           <Text style={[styles.timeText, styles.classText]}>Csc 121</Text>
                           <Text style={[styles.timeText, styles.classPlace]}>Lecture room 3</Text>
                       </View>
                       <View>
                            <Button onPress={()=>playSound()} style={styles.iconNot} transparent>
                                <MaterialIcon  size={30} color="#fff" name='notifications-active'/>
                            </Button>
                       </View>
                </LinearGradient>
            </View>
    );
}
const styles = EStyleSheet.create({
    action:{
        height:"100rem",
        width:"330rem",
        alignSelf: 'center',
        overflow: 'hidden',
        borderRadius:'10rem',
        borderBottomLeftRadius: '0rem',
        borderTopLeftRadius: '0rem',
        marginTop:'20rem'
    },
    gradient:{
        height:"100rem",
        width:"330rem",
        flexDirection:'row'
    },
    eventList:{
        marginTop:'50rem',  
    },
    line:{
      width:"4rem",
      height:"80%",
      backgroundColor:"#fff",
      marginTop:'10rem'
    },
    timeText:{
        fontSize: '24rem', 
        color:"#fff",
        fontFamily:"Itim", 
        padding: "8rem",
    },
    classText:{
        fontSize: '30rem', 
        padding:"5rem"
    },
    classPlace:{
        paddingLeft: '5rem',
    },
    iconNot:{
        marginTop:'20rem',
        marginLeft:'10rem'
    }
   
})
export default DateAction;