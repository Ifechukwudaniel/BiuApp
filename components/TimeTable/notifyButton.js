import React, {useState} from 'react';
import {View, StatusBar, TouchableOpacity, ScrollView} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import { LinearGradient } from 'expo-linear-gradient';
import DateItem from './DateItem'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
import Ripple from 'react-native-material-ripple';
import { Audio } from 'expo-av';
import Header from '../Header'
import { Container,  Button, Text,  } from 'native-base'


const playSound=  async ()=>{
    const soundObject = new Audio.Sound();
    try {
        await soundObject.loadAsync(require('../../assets/sounds/beep.mp3'));
        await soundObject.playAsync();
        } catch (error) {
    }
}
 
const NotifyButton = ({notify, onNotifyPress}) => {
    const [active, setActive] = useState(notify) 
    return (
        <Button onPress={()=>{
            playSound()
            setActive(!active)
            onNotifyPress()
            }
            } style={styles.iconNot} transparent>
             <MaterialIcon  size={30} color="#fff" name={active?'notifications-active':'notifications-off'}/>
         </Button>

    );
}
const styles = EStyleSheet.create({
    iconNot:{
        marginTop:'20rem',
        marginLeft:'10rem'
    }
   
})
export default NotifyButton;