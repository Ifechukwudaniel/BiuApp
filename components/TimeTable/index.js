import React, {useState} from 'react';
import {View, StatusBar, TouchableOpacity, ScrollView} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import { LinearGradient } from 'expo-linear-gradient';
import DateItem from './DateItem'
import Header from '../Header'
import { Container,  Button, Text,  } from 'native-base';
import DateAction from './DateAction'
import { Select } from '@ui-kitten/components';


const data = [
    { text: 'Personal  TimeTable' },
    { text: ' Departmental Timetable' },
  ];
 
const TimeTable = ({navigation}) => {
    const [selectedOption, setSelectedOption] = React.useState(data[0]);
    return (
        <View style={styles.container}>
        <Header  screenName="Time Table" onBackPress= {()=>navigation.push("Home")} />
         <View style={styles.content}>
           <View style={{flexDirection:"row"}}>
           <Text style={styles.dayText}> Today </Text>
           </View>
            <ScrollView style={styles.scrollView} horizontal={true} showsHorizontalScrollIndicator={false} >
                <DateItem date={16} day={0} active/>
                <DateItem date={17} day={1} />
                <DateItem date={18} day={2} />
                <DateItem date={19} day={3} />
                <DateItem date={20} day={4} />
                <DateItem date={21} day={5} />
                <DateItem date={22} day={6} />
            </ScrollView>
           <ScrollView style={styles.eventList}>
                <DateAction onNotifyPress={()=>{}} location="lecture room 3"  course="Csc 121"  startTime="8:00 am" endTime="10:00 am"/>
                <DateAction active={true} notify={true} onNotifyPress={()=>alert("dkd")} location="lecture room 1"  course="Csc 123"  startTime="10:00 am" endTime="12:00 am" />
           </ScrollView>
         </View>
        </View>
    );
}
const styles = EStyleSheet.create({
    container:{
     flex: 1,
    },
    title:{
        fontSize: '23rem', 
        color:"#fff",
        fontFamily:"Itim",
    },
    content:{
        marginTop: '10rem',
    },
    dayText:{
        fontSize: '30rem', 
        color:"#fff",
        fontFamily:"Itim", 
        marginLeft: '10rem',
        marginRight:"10rem"
    },
    scrollView:{
        marginLeft: '20rem',
        marginTop:"20rem",
        flexDirection: 'row',
    },
    eventList:{
        marginTop:'50rem',  
    },  
    option :{
        width:"257rem",
        backgroundColor: "#000",
    }
})
export default TimeTable;