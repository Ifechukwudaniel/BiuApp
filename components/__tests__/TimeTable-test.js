import React from 'react';
import renderer from 'react-test-renderer';

import TimeTable from '../TimeTable';

it(` Time Table renders correctly`, () => {
  const tree = renderer.create( <TimeTable navigation={{}}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
