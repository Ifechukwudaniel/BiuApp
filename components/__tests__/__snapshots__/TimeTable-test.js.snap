// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 4,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007aff",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 2,
                  "paddingRight": undefined,
                  "top": 1,
                }
              }
              uppercase={false}
            >
              Back
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "700",
                  "marginLeft": undefined,
                  "paddingLeft": 4,
                  "paddingTop": 1,
                  "textAlign": "center",
                },
                undefined,
                undefined,
              ]
            }
          >
             
            Time Table
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        />
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
         Today 
      </Text>
    </View>
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4293540153,
                  4288090899,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              16
               
            </Text>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              Mon
               
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279398398,
                  4278923628,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              17
               
            </Text>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              Tue
               
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279398398,
                  4278923628,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              18
               
            </Text>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              Wen
               
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279398398,
                  4278923628,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              19
               
            </Text>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              Thu
               
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279398398,
                  4278923628,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              20
               
            </Text>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              Fri
               
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279398398,
                  4278923628,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              21
               
            </Text>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              Sat
               
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279398398,
                  4278923628,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              22
               
            </Text>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
               
              Sun
               
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </RCTScrollView>
    <RCTScrollView>
      <View>
        <View>
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4279212615,
                  4278351921,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
              >
                8:00 am
              </Text>
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
              >
                10:00 am
              </Text>
            </View>
            <View />
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    undefined,
                    undefined,
                  ]
                }
                uppercase={false}
              >
                Csc 121
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    undefined,
                    undefined,
                  ]
                }
                uppercase={false}
              >
                lecture room 3
              </Text>
            </View>
            <View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 5,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "borderWidth": undefined,
                    "elevation": 0,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": null,
                    "shadowOffset": null,
                    "shadowOpacity": null,
                    "shadowRadius": null,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
        <View>
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4293540153,
                  4288090899,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
              >
                10:00 am
              </Text>
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
              >
                12:00 am
              </Text>
            </View>
            <View />
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    undefined,
                    undefined,
                  ]
                }
                uppercase={false}
              >
                Csc 123
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    undefined,
                    undefined,
                  ]
                }
                uppercase={false}
              >
                lecture room 1
              </Text>
            </View>
            <View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 5,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "borderWidth": undefined,
                    "elevation": 0,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": null,
                    "shadowOffset": null,
                    "shadowOpacity": null,
                    "shadowRadius": null,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </ViewManagerAdapter_ExpoLinearGradient>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
